tests/models/test_python_api.py

```diff
@@ -0,0 +1,180 @@
+import json
```

Comment:
@B-Step62 Can we add test coverage for `pip_requirements_override` with `conda` and `virtualenv` as env managers? Looks like all the cases in this suite set `pip_requirements_override` to `None`

Reply:
@dbczumar sure I will create a follow-up PR for that!

Reply:
Can we add it as part of this PR? Else, how do we know it's safe to merge this PR?

Reply:
I'm fine with doing it here, just tried to make it easy to review. (Btw I tested the functionality for those env managers  in manual E2E test above).